indico/modules/events/layout/clone.py

```diff
@@ -53,3 +57,36 @@ def clone(self, new_event, options):
             db.session.add(new_image)
             db.session.flush()
             logger.info('Added image during event cloning: {}'.format(new_image))
+
+
+class LayoutCloner(EventCloner):
+    def get_options(self):
+        if self.event.getType() != 'conference':
+            return {}
+        return {'layout': (_('Layout settings and menu customization'), True, False)}
+
+    def clone(self, new_event, options):
+        if 'layout' not in options:
+            return
+
+        layout_settings.set_multi(new_event, layout_settings.get_all(self.event))
+        if layout_settings.get(self.event, 'use_custom_menu'):
+            for menu_entry in MenuEntry.get_for_event(self.event):
+                self._copy_menu_entry(menu_entry, new_event)
+            db.session.flush()
+
+    def _copy_menu_entry(self, menu_entry, event):
+        base_columns = {column.key for column in inspect(MenuEntry).column_attrs} - {'id', 'event_id', 'parent_id',
+                                                                                     'page_id'}
+        new_menu_entry = MenuEntry(event_id=event.id, **{col: getattr(menu_entry, col) for col in base_columns})
+        if menu_entry.is_page:
+            new_page = MenuPage(html=menu_entry.page.html)
+            db.session.add(new_page)
```

Comment:
you don't have to add/flush all the time. when you associate the page with the entry, it's added automatically